public/app/features/panel/metrics_panel_ctrl.ts

```diff
@@ -166,6 +166,23 @@ class MetricsPanelCtrl extends PanelCtrl {
       return this.$q.when([]);
     }
 
+    var targets = JSON.stringify(this.panel.targets);
+    var intervalVar = null;
+    _.each(this.templateSrv.variables, function(variable) {
```

Comment:
This feels very hacks,  can you describe what your doing and why, also is there a better way?